src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -199,21 +203,66 @@ void SensorFusion::updateWorld(const SSL_DetectionFrame &ssl_detection_frame)
             friendly_team = createFriendlyTeam(blue_team);
             enemy_team    = createEnemyTeam(yellow_team);
         }
+        updateRobotStatesMap();
     }
 
     if (new_ball_state)
     {
-        if (ball)
+        updateBall(*new_ball_state);
+    }
+}
+
+void SensorFusion::updateBall(TimestampedBallState new_ball_state)
+{
+    if (!ball_states.empty() &&
+        new_ball_state.timestamp() < ball_states.front().timestamp())
+    {
+        throw std::invalid_argument(
+            "Error: Trying to update ball state using a state older then the current state");
+    }
+
+    ball_states.push_front(new_ball_state);
+
+    if (ball)
+    {
+        ball->updateState(new_ball_state);
+    }
+    else
+    {
+        ball = Ball(new_ball_state);
+    }
+}
+
+void SensorFusion::updateRobotStatesMap(const Team &team, TeamHistoryMap &map_to_update)
```

Comment:
The `Team` data type is our abstraction around the filtered data that we pass to the AI. I'm a bit confused about why we're updating the `TeamHistoryMap` with a `Team` here? I would expect to use raw data to update the `TeamHistoryMap`, and then create the `Team`'s from that. 

I'm sure I'm missing something here, just trying to clarify.

Reply:
I think that Team should have a map of id to robotstate, but we're not there yet. It's confusing to have it in this PR, so I'll remove it.